File: Two Sum/two_sum.ts
function twoSum(nums: number[], target: number): number[] {
    let m: Record<number, number> = {}
    for (let i = 0; i < nums.length; i++) {
        if (target - nums[i] in m) {
            return [m[target - nums[i]], i]
        }
        m[nums[i]] = i
    }
    return []
}

function main() {
    let array = [0, 1, 2, 4, 5, 6]
    let target = 5
    console.log(twoSum(array, target))
}
main()

File: Two Sum/two_sum.kt
fun twoSum(nums: Array<Int>, target: Int): Pair<Int, Int> {
  var cand = mutableMapOf<Int, Int>()
  for ((idx, num) in nums.withIndex()) {
      if (target - num in cand) {
        return Pair(cand[target - num]!!, idx)
      }
      cand.set(num, idx)
  }
  return Pair(0, 0)
}

fun main() {
  val arr = arrayOf(1, 2, 3, 4, 5)
  val target = 4
  println(twoSum(arr, target))
}
